Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java

```diff
@@ -251,25 +253,81 @@ private EventIdentificationType getEventIdentificationType(CodedValueType eventI
         return eventIdentification;
     }
 
-    private AuditMessageType.ActiveParticipant getActiveParticipantSource(boolean isRequesting,
+    /**
+     * This method builds ActiveParticipant Source object
+     *
+     * @param target
+     * @param serviceName
+     * @param isRequesting
+     * @param webContextProperties
+     * @return
+     */
+    private ActiveParticipant getActiveParticipantSource(NhinTargetSystemType target, String serviceName,
+        boolean isRequesting,
         Properties webContextProperties) {
 
-        String hostAddress = isRequesting ? getLocalHostAddress() : getRemoteHostAddress(webContextProperties);
-
-        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
-        participant.setUserID(AuditTransformsConstants.ACTIVE_PARTICIPANT_USER_ID_SOURCE);
-        participant.setAlternativeUserID(ManagementFactory.getRuntimeMXBean().getName());
-        participant.setNetworkAccessPointID(hostAddress);
-        participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(hostAddress));
-        participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(
-            AuditTransformsConstants.ACTIVE_PARTICIPANT_ROLE_CODE_SOURCE, null,
-            AuditTransformsConstants.ACTIVE_PARTICIPANT_CODE_SYSTEM_NAME,
-            AuditTransformsConstants.ACTIVE_PARTICIPANT_ROLE_CODE_SOURCE_DISPLAY_NAME));
-        participant.setUserIsRequestor(Boolean.TRUE);
+        ActiveParticipant participant = new ActiveParticipant();
+
+        /* if Retrieve Document Service, Activeparticipant Source and destination is same on both initiator and
```

Comment:
This method should be changed to protected and the DR specific logic moved to an overriding method in the inheriting class (DocRetrieveAuditTransforms).